src/Groups/matrices/forms.jl

```diff
@@ -0,0 +1,461 @@
+import AbstractAlgebra: FieldElem, PolyElem, MPolyElem
+import Hecke: base_ring, defining_polynomial, gram_matrix, radical
+
+export
+    alternating_form,
+    corresponding_bilinear_form,
+    corresponding_quadratic_form,
+    hermitian_form,
+    isalternating_form,
+    isdegenerate,
+    ishermitian_form,
+    isquadratic_form,
+    issingular,
+    issymmetric_form,
+    preserved_quadratic_forms,
+    preserved_sesquilinear_forms,
+    quadratic_form,
+    SesquilinearForm,
+    symmetric_form,
+    witt_index
+
+
+
+# descr is always defined
+# matrix is always defined except when descr="quadratic"; in such a case, at least one of matrix and pol is defined
+# NOTE: the fields ring_iso and mat_iso are always defined if the field X is
+"""
+    SesquilinearForm{T<:RingElem}
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct SesquilinearForm{T<:RingElem}
+   matrix::MatElem{T}
+   descr::Symbol       # quadratic, symmetric, alternating or hermitian
```

Comment:
Perhaps it would be more Julian if these were four subtypes of a common abstract base type... ? But I guess it depends on what you do with these further below...

BTW, in the future one might want to admit other forms, even indefinite, degenerate, isotropic ones...